lane.py

```diff
@@ -1475,61 +1475,87 @@ def modify_search_filter_hook(self, filter):
     def works_for_hits(self, _db, hits):
         """Convert a list of search results into Work objects.
 
+        This works by calling works_for_resultsets() on a list
+        containing a single list of search results.
+
         :param _db: A database connection
         :param hits: A list of Hit objects from ElasticSearch.
         :return: A list of Work or (if the search results include
             script fields), WorkSearchResult objects.
         """
 
-        work_ids = [x.work_id for x in hits]
+        [results] = self.works_for_resultsets(_db, [hits])
+        return results
+
+    def works_for_resultsets(self, _db, resultsets):
+        """Convert a list of lists of Hit objects into a list
+        of lists of Work objects.
+        """
+        from external_search import (
+            Filter,
+            WorkSearchResult,
+        )
 
-        # The simplest way to do this is to create a
-        # DatabaseBackedWorkList that fetches those specific works
+        has_script_fields = None
+        work_ids = set()
+        for resultset in resultsets:
+            for result in resultset:
+                work_ids.add(result.work_id)
+                if has_script_fields is None:
+                    # We don't know whether any script fields were
+                    # included, and now we're in a position to find
+                    # out.
+                    has_script_fields = (
+                        any(
+                            x in result for x in Filter.KNOWN_SCRIPT_FIELDS
+                        )
+                    )
```

Comment:
This is instead of looking at the first result (if there is a first result) after iterating over the list. There's no reason we can't do it _while_ we're iterating over the list.

Reply:
What kind of data is in the script fields? It makes sense to do this here but why only check the first result?

Reply:
The only script field we have right now is "simplified.work_last_update", which calculates the last-update time to use when including a book in an OPDS feed.

We don't actually use the script data here, we just need to check whether the script field is present so we can know to return `WorkSearchResult`s instead of `Work`s. (If we just return `Work`s then the script data is lost.) 

The script data is either present on all the results, or not present at all, so checking the first result is enough to answer that question.